core/src/processing/feeder.cc

```diff
@@ -64,10 +65,18 @@ feeder::feeder(std::string const& name,
  *  Destructor.
  */
 feeder::~feeder() {
-  if (!_should_exit) {
-    _should_exit = true;
-    if (_thread.joinable())
+  exit();
+}
+
+void feeder::exit() {
+  std::unique_lock<std::mutex> lock(_started_m);
+  if (_started) {
```

Comment:
I feel like "started" is the opposite boolean of "shouldExit". What do you think about it ?